### ISSUE TITLE
Contract Deployment Failure with the latest update on Testnet.

### ISSUE THREAD

**Issue by rachit2501** [harmony-one/harmony#3450]
**Describe the bug**
Contract deployment are failing using truffle. Receiving a "gasLimit of undefined" error.

**To Reproduce**
Steps to reproduce:
1. Pick up any truffle projects and set it up to deploy to harmony.
2. Try deploying a smart contract on testnet,

**Expected behavior**
Contract should deploy without any issue.

**Screenshots**
![Screenshot from 2020-11-09 21:14:10](https://user-images.githubusercontent.com/43861614/98564420-605b3a00-22d2-11eb-8b8c-51b4b461bfb8.png)

**Environment (please complete the following information):**
 - OS: [Ubuntu 20.04]

**Comment by peekpi** [comment]
Can you share the truffle project?

To port a truffle project from ethereum to harmony, you can try the following steps:
1. Create a '.env' file that supplies harmony node URL, private key/mnemonic.
    1. example .env
        ```bash
        #TESTNET
        #Account: one18t4yj4fuutj83uwqckkvxp9gfa0568uc48ggj7
        export TESTNET_PRIVATE_KEY='01F903CE0C960FF3A9E68E80FF5FFC344358D80CE1C221C3F9711AF07F83A3BD'
        export TESTNET_MNEMONIC='urge clog right example dish drill card maximum mix bachelor section select' 
        
        export TESTNET_0_URL='https://api.s0.b.hmny.io'
        export TESTNET_1_URL='https://api.s1.b.hmny.io'
        
        #MAINNET
        #Please replace MAINNET_PRIVATE_KEY and MAINNET_MNEMONIC with your own!
        #Account: one18t4yj4fuutj83uwqckkvxp9gfa0568uc48ggj7
        export MAINNET_PRIVATE_KEY='01F903CE0C960FF3A9E68E80FF5FFC344358D80CE1C221C3F9711AF07F83A3BD'
        export MAINNET_MNEMONIC='urge clog right example dish drill card maximum mix bachelor section select' 
        export MAINNET_0_URL='https://api.s0.t.hmny.io'
        
        export GAS_LIMIT=80000000
        export GAS_PRICE=1000000000
        ```
    2. execute `source .env`
2. Modify the `truffle-config.js` to add a Harmony Network provider. Here is a sample `truffle_config.js` for the Harmony Network.
    ```js
    const { TruffleProvider } = require('@harmony-js/core')
    
    //Local
    const local_mnemonic = process.env.LOCAL_MNEMONIC
    const local_private_key = process.env.LOCAL_PRIVATE_KEY
    const local_url = process.env.LOCAL_0_URL;
    
    //Testnet
    const testnet_mnemonic = process.env.TESTNET_MNEMONIC
    const testnet_private_key = process.env.TESTNET_PRIVATE_KEY
    const testnet_url = process.env.TESTNET_0_URL
    
    //Mainnet
    const mainnet_mnemonic = process.env.MAINNET_MNEMONIC
    const mainnet_private_key = process.env.MAINNET_PRIVATE_KEY
    const mainnet_url = process.env.MAINNET_0_URL;
    
    //GAS - Currently using same GAS accross all environments
    gasLimit = process.env.GAS_LIMIT
    gasPrice = process.env.GAS_PRICE
    
    module.exports = {
      networks: {
        testnet: {
          network_id: '2', // Any network (default: none)
          provider: () => {
            const truffleProvider = new TruffleProvider(
              testnet_url,
              { memonic: testnet_mnemonic },
              { shardID: 0, chainId: 2 },
              { gasLimit: gasLimit, gasPrice: gasPrice},
            );
            const newAcc = truffleProvider.addByPrivateKey(testnet_private_key);
            truffleProvider.setSigner(newAcc);
            return truffleProvider;
          },
        },
        mainnet0: {
          network_id: '1', // Any network (default: none)
          provider: () => {
            const truffleProvider = new TruffleProvider(
              mainnet_url,
              { memonic: mainnet_mnemonic },
              { shardID: 0, chainId: 1 },
              { gasLimit: gasLimit, gasPrice: gasPrice },
            );
            const newAcc = truffleProvider.addByPrivateKey(mainnet_private_key);
            truffleProvider.setSigner(newAcc);
            return truffleProvider;
          },
        },
      },
    }
    ```
3. add harmony SDK: `yarn add @harmony-js/core tslib`
4. deploy contracts to testnet `truffle --network=local deploy --skip-dry-run`

**Comment by peekpi** [comment]
I create an example truffle project: https://github.com/harmony-one/metacoin-box

**Comment by rachit2501** [comment]
The code base i am using was working perfectly fine a few days back. Here is my config:

Truffle.config.js:
```

require("dotenv").config();
const { TruffleProvider } = require("@harmony-js/core");
//Local
const local_mnemonic = process.env.LOCAL_MNEMONIC;
const local_private_key = process.env.LOCAL_PRIVATE_KEY;
const local_url = process.env.LOCAL_0_URL;
//Testnet
const testnet_mnemonic = process.env.TESTNET_MNEMONIC;
const testnet_private_key = process.env.TESTNET_PRIVATE_KEY;
const testnet_url = process.env.TESTNET_0_URL;
//Mainnet
const mainnet_mnemonic = process.env.MAINNET_MNEMONIC;
const mainnet_private_key = process.env.MAINNET_PRIVATE_KEY;
const mainnet_url = process.env.MAINNET_0_URL;

//GAS - Currently using same GAS accross all environments
gasLimit = process.env.GAS_LIMIT;
gasPrice = process.env.GAS_PRICE;

module.exports = {
  networks: {
    local: {
      network_id: "2", // Any network (default: none)
      provider: () => {
        const truffleProvider = new TruffleProvider(
          local_url,
          { memonic: local_mnemonic },
          { shardID: 0, chainId: 2 },
          { gasLimit: gasLimit, gasPrice: gasPrice }
        );
        const newAcc = truffleProvider.addByPrivateKey(local_private_key);
        truffleProvider.setSigner(newAcc);
        return truffleProvider;
      },
    },
    testnet: {
      network_id: "2", // Any network (default: none)
      provider: () => {
        const truffleProvider = new TruffleProvider(
          testnet_url,
          { memonic: testnet_mnemonic },
          { shardID: 0, chainId: 2 },
          { gasLimit: gasLimit, gasPrice: gasPrice }
        );
        const newAcc = truffleProvider.addByPrivateKey(testnet_private_key);
        truffleProvider.setSigner(newAcc);
        return truffleProvider;
      },
    },
    mainnet0: {
      network_id: "1", // Any network (default: none)
      provider: () => {
        const truffleProvider = new TruffleProvider(
          mainnet_url,
          { memonic: mainnet_mnemonic },
          { shardID: 0, chainId: 1 },
          { gasLimit: gasLimit, gasPrice: gasPrice }
        );
        const newAcc = truffleProvider.addByPrivateKey(mainnet_private_key);
        truffleProvider.setSigner(newAcc);
        return truffleProvider;
      },
    },
  },


  mocha: {
    // timeout: 100000
  },

  compilers: {
    solc: {
      version: "0.5.8",
    },
  },
};

```
.env is same as yours.

- while deploying contracts, getting truffle --network testnet --reset.

Expected: Contract Deploying easily ( like a few days back )
Outcome: Error mentioned above.

Note: 1) I have verified that I am using the latest version of the sdk.
         2) There is another similar truffle config I have, with sol 0.7.0 version and receiving same error after the upgrade to the chain.

**Comment by gupadhyaya** [comment]
I also cannot reproduce this error. @rachit2501 better to push your repo to GitHub and share?

**Comment by rachit2501** [comment]
@gupadhyaya  Try this: https://github.com/rachit2501/Smart-Contract-Demo . This was the config which was working perfectly like 2 weeks ago. The configuration is send above was also working at that time too. Both of them showing the same problem.

**Comment by peekpi** [comment]
> I also cannot reproduce this error. @rachit2501 better to push your repo to GitHub and share?

This issue can be reproduced in the latest truffle(v5.1.52), but don't know what caused it.

**Comment by peekpi** [comment]
It's an issue of JSSDK, has fixed by pr: harmony-one/sdk/pull/66. 
You can use an old Truffle(v5.1.33) until the new JSSDK is published.
